tests/Microsoft.Identity.Test.Unit/PublicApiTests/OBOTests.cs

```diff
@@ -240,23 +240,44 @@ public async Task InitiateLongRunningProcessWithExistingKeyAndToken_TestAsync()
                 MsalAccessTokenCacheItem cachedAccessToken = cca.UserTokenCacheInternal.Accessor.GetAllAccessTokens().Single();
                 MsalRefreshTokenCacheItem cachedRefreshToken = cca.UserTokenCacheInternal.Accessor.GetAllRefreshTokens().Single();
                 Assert.AreEqual(oboCacheKey, cachedAccessToken.OboCacheKey);
                 Assert.AreEqual(oboCacheKey, cachedRefreshToken.OboCacheKey);
-                userCacheAccess.AssertAccessCounts(0, 1);
+                Assert.AreEqual(CacheRefreshReason.NoCachedAccessToken, result.AuthenticationResultMetadata.CacheRefreshReason);
+                userCacheAccess.AssertAccessCounts(1, 1);
+
+                // Initiate another process using the same cache key and assertion
+                // Returns unexpired access token which matches cache key and assertion
+                result = await cca.InitiateLongRunningProcessInWebApi(TestConstants.s_scope, TestConstants.DefaultAccessToken, ref oboCacheKey)
+                    .ExecuteAsync().ConfigureAwait(false);
+
+                Assert.AreEqual(TestConstants.ATSecret, result.AccessToken);
+                Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+                cachedAccessToken = cca.UserTokenCacheInternal.Accessor.GetAllAccessTokens().Single();
+                cachedRefreshToken = cca.UserTokenCacheInternal.Accessor.GetAllRefreshTokens().Single();
+                Assert.AreEqual(oboCacheKey, cachedAccessToken.OboCacheKey);
+                Assert.AreEqual(oboCacheKey, cachedRefreshToken.OboCacheKey);
+                Assert.AreEqual(TestConstants.ATSecret, cachedAccessToken.Secret);
+                Assert.AreEqual(CacheRefreshReason.NotApplicable, result.AuthenticationResultMetadata.CacheRefreshReason);
+                userCacheAccess.AssertAccessCounts(2, 1);
+
+                TokenCacheHelper.ExpireAllAccessTokens(cca.UserTokenCacheInternal);
 
                 AddMockHandlerAadSuccess(httpManager,
-                    responseMessage: MockHelpers.CreateSuccessTokenResponseMessage(accessToken: TestConstants.ATSecret2));
+                    responseMessage: MockHelpers.CreateSuccessTokenResponseMessage(accessToken: TestConstants.ATSecret2, refreshToken: TestConstants.RTSecret2),
+                    expectedPostData: new Dictionary<string, string> { 
+                        { OAuth2Parameter.RequestedTokenUse, OAuth2RequestedTokenUse.OnBehalfOf }, { OAuth2Parameter.GrantType, OAuth2GrantType.JwtBearer } });
 
-                //Initiate another process using the same cache key
-                //MSAL should ignore the token in the cache, fetch a new token and overrite the existing one
+                // Initiate another process using the same cache key and assertion
+                // Access token is expired, uses assertion, not a refresh token, to retrieve new tokens from AAD
```

Comment:
Not sure this is the right call or if I am missing something. If the AT is expired and the same OBO key and assertion is used, AAD wouldnt return a new token because the original user assertion would have expired along with the cached OBO AT. So, in order for this to work, wouldnt we need to use the RT of the expired AT that has the same OBO key and assertion? Tenically it will still be possible to refresh with the long running OBO. Im not saying that we should make the Initiate method use the RT. but I dont think AAD will return a new token in this case as the test suggests.

Reply:
Now that I think about this some more, would't this break a customer who is using the initiate method over and over again once the original assertion has expired?

Reply:
> Now that I think about this some more, would't this break a customer who is using the initiate method over and over again once the original assertion has expired?

Yes, if they upgrade from MSAL 4.50 or below. But as soon as they upgrade to 4.51, then it would break since Initiate would go to AAD each time with expired assertion (which will throw).

Reply:
Hmm, yea I think I can update to also check for RT with the same assertion - if found use RT to refresh, otherwise use request assertion?

In the scenario when the cached tokens are revoked, we'd expect Initiate to be called with a new / uncached assertion.